src/view/com/composer/videos/state.ts

```diff
@@ -0,0 +1,49 @@
+import {useState} from 'react'
+import {ImagePickerAsset} from 'expo-image-picker'
+import {msg} from '@lingui/macro'
+import {useLingui} from '@lingui/react'
+import {useMutation} from '@tanstack/react-query'
+
+import {compressVideo} from '#/lib/media/video/compress'
+
+export function useVideoState({setError}: {setError: (error: string) => void}) {
+  const {_} = useLingui()
+  const [progress, setProgress] = useState(0)
+
+  const {mutate, data, isPending, isError, reset, variables} = useMutation({
+    mutationFn: async (asset: ImagePickerAsset) => {
+      if (asset.fileSize) {
+        console.log(
+          'uncompressed size',
+          (asset.fileSize / 1024 / 1024).toFixed(2) + 'mb',
+        )
```

Comment:
log (sry i know it's a draft just want to make sure i get these in while im here 😅)